src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -77,7 +77,7 @@ internal static class MetadataUtilities
                 }
                 if (modulesByName == null)
                 {
-                    modulesByName = new Dictionary<string, ModuleMetadata>(); // Requires case-insensitive comparison?
+                    modulesByName = new Dictionary<string, ModuleMetadata>(StringComparer.OrdinalIgnoreCase); // Requires case-insensitive comparison
```

Comment:
Can we test this change? (See `ReferencedModulesTests` for an example that calls `MetadataUtilities.MakeAssemblyReferences`.)